doc/upload.md

```diff
@@ -0,0 +1,153 @@
+# CDI Upload User Guide
+The purpose of this document is to show how to upload a VM disk image on your local system to a PersistentVolumeClaim in Kubernetes.
```

Comment:
Might want to tweak the wording on this a little bit.  @awels did a really nice job IMO on the DV doc, it would be great to keep that sort of format and level of detail if we can.  See https://github.com/kubevirt/containerized-data-importer/pull/466/commits/8c98b0e01f01d3faadbcb144227b9abf816feddc for what I'm thinking.

Having the description of what's going on, why it's useful and the flow seems like a really nice template for docs when a user wants to read about features.

Reply:
Since DataVolumes are a higher level construct, it totally makes sense to go into the "what" and the "why."  To me, the what and why of upload don't require much explanation and that is the reason I went with "user guide" approach.  What am I missing?  Maybe I've been in the weeds with this stuff for to long.